pkgs/build-support/writers/scripts.nix

```diff
@@ -69,10 +84,15 @@ rec {
         ${check} $out
       ''}
       chmod +x $out
+
+      # Relocate executable if path was specified instead of name.
+      # Only in this case wrapProgram is applied, as it wouldn't work with a
+      #   single executable file under $out.
       ${optionalString (types.path.check nameOrPath) ''
         mv $out tmp
         mkdir -p $out/$(dirname "${nameOrPath}")
         mv tmp $out/${nameOrPath}
+        wrapProgram $out/${nameOrPath} ''${makeWrapperArgs[@]}
```

Comment:
shouldn't this check if `makeWrapperArgs` is empty before wrapping

Reply:
done + added test checking that no empty wrappers are created